pkg/node/node.go

```diff
@@ -309,5 +309,20 @@ func (n *Node) Marshal() ([]byte, error) {
 
 // Unmarshal parses the JSON byte slice and updates the node receiver
 func (n *Node) Unmarshal(data []byte) error {
-	return json.Unmarshal(data, n)
+	// Backup private fields
+	cluster := n.cluster
+	dev := n.dev
+
+	// Re-initialize the node object as Unmarshal() may not overwrite all
+	// fields.
+	*n = Node{}
```

Comment:
Why don't you pass a new node to the json.Unmarshall and only if there is not error you assign the new node to the receiver?

Reply:
Could also explicitly initialize all the public fields instead of backing up the private fields.